src/utils/searchViewHelper.ts

```diff
@@ -12,79 +13,65 @@ import {
+    store.dispatch(
+      setInputError(
+        'Search must be a 12- or 40-character hash, or email address',
+      ),
+    );
+  }
+};
 
-  handleChangeDropdown(event) {
-    const repository = event.target.id;
-    const search = store.getState().search.searchValue;
+const handleChangeDropdown = (event: MouseEvent) => {
+  const repository = (event.target as Element).id as Repository['name'];
 
-    // Update state with selected repository
-    store.dispatch(updateRepository(repository));
-    // If repository is selected after search value is input, fetch search results
-    if (search !== '') {
-      searchViewHelper.searchByRevisionOrEmail(repository, search);
-    } else {
-      // Fetch 10 most recent revisions when repository changes
-      store.dispatch(fetchRecentRevisions(repository));
-    }
-    store.dispatch(updateSearchIsFocused(true));
-  },
+  // Update state with selected repository
+  store.dispatch(updateRepository(repository));
 
-  handleChangeSearch(event) {
-    const search = event.target.value;
-    const { repository } = store.getState().search;
-    store.dispatch(updateSearchValue(search));
-    store.dispatch(updateSearchResults([]));
-    store.dispatch(clearInputError());
+  // Fetch 10 most recent revisions when repository changes
+  store.dispatch(fetchRecentRevisions(repository));
+};
 
-    const idleTime = 500;
-    const onTimeout = () => {
-      searchViewHelper.searchByRevisionOrEmail(repository, search);
-    };
+const handleChangeSearch = (
+  event: FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+) => {
+  const search = event.currentTarget.value;
+  const { repository } = store.getState().search;
+  store.dispatch(updateSearchValue(search));
+  store.dispatch(updateSearchResults([]));
+  store.dispatch(clearInputError());
 
-    // Clear any existing timer whenever user types
-    if (timeout) clearTimeout(timeout);
+  const idleTime = 500;
+  const onTimeout = () => {
+    searchByRevisionOrEmail(repository, search);
+  };
 
-    // If search input is cleared, clear results
-    if (search === '') {
-      store.dispatch(updateSearchResults([]));
-    } else {
-      // Submit API call 500ms after user stops typing
-      timeout = setTimeout(onTimeout, idleTime);
-    }
-  },
+  // Clear any existing timer whenever user types
+  if (timeout) clearTimeout(timeout);
 
-  handleClickOutsideInput(event) {
-    if (
-      event.target.matches(
-        '#search-revision-input, .search-revision, .search-revision *',
-      )
-    ) {
-      store.dispatch(updateSearchIsFocused(true));
-    } else {
-      store.dispatch(updateSearchIsFocused(false));
-    }
-  },
+  // If search input is cleared, clear results
+  if (search === '') {
+    store.dispatch(updateSearchResults([]));
+  } else {
+    // Submit API call 500ms after user stops typing
+    timeout = setTimeout(onTimeout, idleTime);
+  }
 };
 
-export default searchViewHelper;
+export { handleChangeDropdown, handleChangeSearch };
```

Comment:
optional nit: I believe you could do directly `export const searchByRevisionOrEmail = ...`